lib/ansible/galaxy/dependency_resolution/dataclasses.py

```diff
@@ -418,6 +439,19 @@ def from_requirement_dict(cls, collection_req, art_mgr, validate_signature_optio
                 'index server.',
             )
 
+        if req_type != 'git' and req_version != '*':
+            for segment in req_version.split(','):
+                if segment[:2] in OP_MAP:
+                    segment = segment[2:]
+                elif segment[0] in OP_MAP:
+                    segment = segment[1:]
+                try:
+                    SemanticVersion.from_loose_version(LooseVersion(segment))
+                except ValueError:
+                    raise AnsibleError(
+                        f"The {req_name} 'version' key is not a valid version."
+                    )
```

Comment:
Could you move the for-loop plus check into `versioning.py`? It already has related code, and it'd be better semantically to keep it close.
Additionally, it would reduce the level of nesting here. This method is complex enough. Just compare with this:

```suggestion
            try:
                assert_requrements_valid(req_version)
            except ValueError:
                raise AnsibleError(
                    f"The {req_name} 'version' key is not a valid version: "
                    f"'{segment}' part of '{req_version}' does not follow SemVer."
                )
```